tests/std/include/range_algorithm_support.hpp

```diff
@@ -122,6 +122,26 @@ namespace test {
             ptr_ = s.peek();
         }
         // clang-format on
+
+        friend constexpr boolish operator==(sentinel const s, Element* const ptr) noexcept {
+            return {s.ptr_ == ptr};
+        }
+        friend constexpr boolish operator==(Element* const ptr, sentinel const s) noexcept {
+            return {s.ptr_ == ptr};
+        }
+        friend constexpr boolish operator!=(sentinel const s, Element* const ptr) noexcept {
+            return !(s == ptr);
+        }
+        friend constexpr boolish operator!=(Element* const ptr, sentinel const s) noexcept {
+            return !(s == ptr);
+        }

```

Comment:
I believe (hope) those 3 are superfluous due to C++20 operator rewriting?

Reply:
Also I am confused. Do we use `east const` or `const west`?

Reply:
`test::iterator` and `test::sentinel` have to implement their own comparisons because they are ill-behaved and return `boolish`. C++20 comparison expression rewrites are only activated when the target of the rewrite has type `bool`. (I should say "intentionally ill-behaved" here since the point is to ensure that the algorithms work with ill-behaved user types.)

Reply:
> Also I am confused. Do we use `east const` or `const west`?

Internal consistency within a file is paramount, and this header was one of the last things I wrote before deciding to acquiesce to the prevalent `const`-west style in the test suite. I will eventually change this to `const`-west when we reach a point where I can make a global change like that in this header without conflicting with any in-flight PRs.

Reply:
Thanks for the explanation. 

I am (fortunately?) not yet accustomed with all those hairy corner cases.